File: Behavioral/Lab17/C#/Directors/EstateDirector.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Builders;
using C_.Entities;

namespace C_.Directors
{
    public abstract class EstateDirector
    {
        protected EstateBuilder _estateBuilder;
        public EstateDirector(EstateBuilder estateBuilder){
            _estateBuilder = estateBuilder;
        }

        public abstract RealEstate BuildDefault();
    }
}

File: Behavioral/Lab22/Typescript/src/Directors/estate_director.ts
import { EstateBuilder } from "../Builders/estate_builder";
import { RealEstate } from "../Entities/real_estate";

export abstract class EstateDirector {
    protected _estateBuilder: EstateBuilder;

    constructor(estateBuilder: EstateBuilder) {
        this._estateBuilder = estateBuilder;
    }
    public abstract BuildDefault():RealEstate;
}